uk.co.compucorp.civicrm.hrcomments/CRM/HRComments/BAO/Comment.php

```diff
@@ -146,6 +145,25 @@ private static function validateCommentCreatedAtDate($params) {
   }
 
   /**
+   * Validates that the created_at parameter should not be allowed
+   * to be changed during an update.
+   *
+   * @param array $params
+   *   The params array received by the create method
+   *
+   * @throws \CRM_HRComments_Exception_InvalidCommentException
+   */
+  private static function validateCommentCreatedAtDateDuringAnUpdate($params) {
+    if (isset($params['id']) && !empty($params['created_at'])) {
```

Comment:
So here, the docblock says something but the method is doing another thing. The doc says the `created_at` cannot be changed (that is, the value was 2016-01-01 and now they are trying to change it to 2016-01-02). But what the method is doing is, even if the given date is the same as the current one, it will throw an exception.

I think the better option is to load the comment with the given ID and then compare if the dates are different. The reason is that, depending on how the frontend code works, it's probably easier for them to always pass the created_at value, even if it has not changed. So doing this would avoid unexpected errors